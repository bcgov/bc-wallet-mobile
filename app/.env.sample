MEDIATOR_URL=https://f326-207-194-65-204.ngrok.io?c_i=eyJAdHlwZSI6ICJodHRwczovL2RpZGNvbW0ub3JnL2Nvbm5lY3Rpb25zLzEuMC9pbnZpdGF0aW9uIiwgIkBpZCI6ICI2MjQ0ZThiNS0wNWYzLTRhYWItYjM1Yy1lYWVlMWNmZTAyM2MiLCAicmVjaXBpZW50S2V5cyI6IFsiQ3lqM1BHRUJzQ3RyUGFtTTQyRngza3BlYmR2QWdNd1lGejlFS3RmNnlUN3giXSwgImxhYmVsIjogIk1lZGlhdG9yIiwgInNlcnZpY2VFbmRwb2ludCI6ICJodHRwczovL2YzMjYtMjA3LTE5NC02NS0yMDQubmdyb2suaW8ifQ==

# Push notification variables
MEDIATOR_USE_PUSH_NOTIFICATIONS=false
MEDIATOR_LABEL=Mediator