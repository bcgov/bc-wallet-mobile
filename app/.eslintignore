node_modules
__tests__
__mocks__
ios
android

# generated by bob
lib/
bifold-app/
generate-app-package-json.js