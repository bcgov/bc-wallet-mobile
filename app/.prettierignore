package-lock.json
app.json
ios/
android/
node_modules/
generate-app-package-json.js
src/configs/ledgers/indy/ledgers.json
.gitignore
