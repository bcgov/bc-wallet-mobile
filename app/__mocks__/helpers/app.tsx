import 'reflect-metadata'
import React, { PropsWithChildren, useMemo } from 'react'

import { container } from 'tsyringe'
import { MainContainer, ContainerProvider } from '@hyperledger/aries-bifold-core'

export const BasicAppContext: React.FC<PropsWithChildren> = ({ children }) => {
  const context = useMemo(() => new MainContainer(container.createChildContainer()).init(), [])
  return (
    <ContainerProvider value={context}>{children}</ContainerProvider>
  )
}