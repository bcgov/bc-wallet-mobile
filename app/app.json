{
  "name": "BCWallet",
  "displayName": "BC Wallet"
}
