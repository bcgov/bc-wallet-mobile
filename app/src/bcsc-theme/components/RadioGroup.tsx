import { testIdForAccessabilityLabel, useTheme } from '@bifold/core'
import React from 'react'
import { StyleSheet, View, ViewStyle } from 'react-native'
import { RadioButton } from './RadioButton'

interface RadioOption<T> {
  label: string
  value: T
  disabled?: boolean
}

interface RadioGroupProps<T> {
  options: RadioOption<T>[]
  selectedValue?: T
  onValueChange: (value: T) => void
  testID: string
  style?: ViewStyle
}

export const RadioGroup = <T,>({ options, selectedValue, onValueChange, testID, style = {} }: RadioGroupProps<T>) => {
  const { Spacing } = useTheme()

  const styles = StyleSheet.create({
    optionsContainer: {
      paddingBottom: Spacing.xs,
    },
    separator: {
      height: Spacing.sm,
    },
  })

  return (
    <View testID={testID} style={style}>
      <View style={styles.optionsContainer}>
        {options.map((option, index) => (
          <React.Fragment key={option.label}>
            {index > 0 && <View style={styles.separator} />}
            <RadioButton<T>
              label={option.label}
              value={option.value}
              selectedValue={selectedValue}
              onValueChange={onValueChange}
              disabled={option.disabled}
              testID={`${testID}-option-${testIdForAccessabilityLabel(option.label)}`}
            />
          </React.Fragment>
        ))}
      </View>
    </View>
  )
}
