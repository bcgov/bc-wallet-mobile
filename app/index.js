/* eslint-disable import/no-extraneous-dependencies */
/**
 * @format
 */
import 'react-native-gesture-handler'
// remove these when updated to react-native 0.65.0
import '@formatjs/intl-getcanonicallocales/polyfill'
import '@formatjs/intl-locale/polyfill'
import '@formatjs/intl-pluralrules/polyfill'
import '@formatjs/intl-pluralrules/locale-data/en' // locale-data for en
import '@formatjs/intl-displaynames/polyfill'
import '@formatjs/intl-displaynames/locale-data/en' // locale-data for en
import '@formatjs/intl-listformat/polyfill'
import '@formatjs/intl-listformat/locale-data/en' // locale-data for en
import '@formatjs/intl-numberformat/polyfill'
import '@formatjs/intl-numberformat/locale-data/en' // locale-data for en
import '@formatjs/intl-relativetimeformat/polyfill'
import '@formatjs/intl-relativetimeformat/locale-data/en' // locale-data for en
import '@formatjs/intl-datetimeformat/polyfill'
import '@formatjs/intl-datetimeformat/locale-data/en' // locale-data for en
import '@formatjs/intl-datetimeformat/add-all-tz' // Add ALL tz data

import { NavigationContainer } from '@react-navigation/native'
import React from 'react'
import { AppRegistry } from 'react-native'

import App from './App'
import { name as appName } from './app.json'
import bcwallet from './src/'
const { theme } = bcwallet

const { ColorPallet } = theme

const navigationTheme = {
  dark: true,
  colors: {
    primary: ColorPallet.brand.primary,
    background: ColorPallet.brand.primaryBackground,
    card: ColorPallet.brand.primary,
    text: ColorPallet.grayscale.white,
    border: ColorPallet.grayscale.white,
    notification: ColorPallet.grayscale.white,
  },
}

const Base = () => {
  return (
    <NavigationContainer theme={navigationTheme}>
      <App />
    </NavigationContainer>
  )
}
AppRegistry.registerComponent(appName, () => Base)
