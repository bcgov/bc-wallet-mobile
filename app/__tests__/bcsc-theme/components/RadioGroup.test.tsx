import { fireEvent, render, waitFor } from '@testing-library/react-native'
import React from 'react'

import { BasicAppContext } from '../../../__mocks__/helpers/app'
import { RadioGroup } from '../../../src/bcsc-theme/components/RadioGroup'

describe('RadioGroup Component', () => {
  const defaultOptions = [
    { label: 'Option 1', value: 'option1' },
    { label: 'Option 2', value: 'option2' },
    { label: 'Option 3', value: 'option3' },
  ]

  const defaultProps = {
    options: defaultOptions,
    onValueChange: jest.fn(),
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('Rendering', () => {
    test('renders correctly with default props', () => {
      const tree = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} />
        </BasicAppContext>
      )

      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })

    test('renders correctly with title', () => {
      const tree = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} title="Choose an option" />
        </BasicAppContext>
      )

      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })

    test('renders correctly with selected value', () => {
      const tree = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} selectedValue="option2" />
        </BasicAppContext>
      )

      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })

    test('renders correctly when disabled', () => {
      const tree = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} disabled={true} />
        </BasicAppContext>
      )

      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })

    test('renders correctly with mixed disabled options', () => {
      const optionsWithDisabled = [
        { label: 'Option 1', value: 'option1' },
        { label: 'Option 2 (Disabled)', value: 'option2', disabled: true },
        { label: 'Option 3', value: 'option3' },
      ]

      const tree = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} options={optionsWithDisabled} />
        </BasicAppContext>
      )

      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })

    test('displays all option labels', () => {
      const { getByText } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} />
        </BasicAppContext>
      )

      expect(getByText('Option 1')).toBeTruthy()
      expect(getByText('Option 2')).toBeTruthy()
      expect(getByText('Option 3')).toBeTruthy()
    })

    test('displays title when provided', () => {
      const { getByText } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} title="Select your choice" />
        </BasicAppContext>
      )

      expect(getByText('Select your choice')).toBeTruthy()
    })

    test('does not display title when not provided', () => {
      const { queryByText } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} />
        </BasicAppContext>
      )

      // Should not find any title text
      expect(queryByText('Select your choice')).toBeFalsy()
    })
  })

  describe('Interaction', () => {
    test('calls onValueChange when an option is selected', () => {
      const mockOnValueChange = jest.fn()
      const { getByText } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} onValueChange={mockOnValueChange} />
        </BasicAppContext>
      )

      fireEvent.press(getByText('Option 2'))

      expect(mockOnValueChange).toHaveBeenCalledWith('option2')
      expect(mockOnValueChange).toHaveBeenCalledTimes(1)
    })

    test('calls onValueChange with correct value for each option', () => {
      const mockOnValueChange = jest.fn()
      const { getByText } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} onValueChange={mockOnValueChange} />
        </BasicAppContext>
      )

      fireEvent.press(getByText('Option 1'))
      expect(mockOnValueChange).toHaveBeenCalledWith('option1')

      fireEvent.press(getByText('Option 3'))
      expect(mockOnValueChange).toHaveBeenCalledWith('option3')

      expect(mockOnValueChange).toHaveBeenCalledTimes(2)
    })

    test('does not call onValueChange when group is disabled', () => {
      const mockOnValueChange = jest.fn()
      const { getByText } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} onValueChange={mockOnValueChange} disabled={true} />
        </BasicAppContext>
      )

      fireEvent.press(getByText('Option 1'))

      expect(mockOnValueChange).not.toHaveBeenCalled()
    })

    test('does not call onValueChange for individually disabled options', () => {
      const mockOnValueChange = jest.fn()
      const optionsWithDisabled = [
        { label: 'Option 1', value: 'option1' },
        { label: 'Option 2', value: 'option2', disabled: true },
        { label: 'Option 3', value: 'option3' },
      ]

      const { getByText } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} options={optionsWithDisabled} onValueChange={mockOnValueChange} />
        </BasicAppContext>
      )

      // Press enabled option - should work
      fireEvent.press(getByText('Option 1'))
      expect(mockOnValueChange).toHaveBeenCalledWith('option1')

      // Press disabled option - should not work
      fireEvent.press(getByText('Option 2'))
      expect(mockOnValueChange).toHaveBeenCalledTimes(1) // Still just the first call

      // Press another enabled option - should work
      fireEvent.press(getByText('Option 3'))
      expect(mockOnValueChange).toHaveBeenCalledWith('option3')
      expect(mockOnValueChange).toHaveBeenCalledTimes(2)
    })
  })

  describe('Accessibility Selection State', () => {
    test('shows correct selection state for each option', () => {
      const { getByTestId } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} selectedValue="option2" testID="radio-group" />
        </BasicAppContext>
      )

      // Check that the correct option shows as selected via accessibility state
      const option1 = getByTestId('radio-group-option-option1')
      const option2 = getByTestId('radio-group-option-option2')
      const option3 = getByTestId('radio-group-option-option3')

      expect(option1.props.accessibilityState.selected).toBe(false)
      expect(option2.props.accessibilityState.selected).toBe(true)
      expect(option3.props.accessibilityState.selected).toBe(false)
    })

    test('updates selection state when value changes', () => {
      const mockOnValueChange = jest.fn()
      let selectedValue = 'option1'

      const TestComponent = () => (
        <BasicAppContext>
          <RadioGroup
            {...defaultProps}
            selectedValue={selectedValue}
            onValueChange={(value) => {
              selectedValue = value
              mockOnValueChange(value)
            }}
            testID="radio-group"
          />
        </BasicAppContext>
      )

      const { getByText, getByTestId, rerender } = render(<TestComponent />)

      // Initially option1 should be selected
      expect(getByTestId('radio-group-option-option1').props.accessibilityState.selected).toBe(true)

      // Select option2
      fireEvent.press(getByText('Option 2'))

      // Re-render with new selected value
      selectedValue = 'option2'
      rerender(<TestComponent />)

      expect(getByTestId('radio-group-option-option2').props.accessibilityState.selected).toBe(true)
      expect(getByTestId('radio-group-option-option1').props.accessibilityState.selected).toBe(false)
    })
  })

  describe('Accessibility', () => {
    test('has correct testID when provided', () => {
      const { getByTestId } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} testID="custom-radio-group" />
        </BasicAppContext>
      )

      expect(getByTestId('custom-radio-group')).toBeTruthy()
    })

    test('generates correct testIDs for individual options', () => {
      const { getByTestId } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} testID="test-group" />
        </BasicAppContext>
      )

      expect(getByTestId('test-group-option-option1')).toBeTruthy()
      expect(getByTestId('test-group-option-option2')).toBeTruthy()
      expect(getByTestId('test-group-option-option3')).toBeTruthy()
    })

    test('uses default testID for options when no testID provided', () => {
      const { getByTestId } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} />
        </BasicAppContext>
      )

      expect(getByTestId('radioGroup-option-option1')).toBeTruthy()
      expect(getByTestId('radioGroup-option-option2')).toBeTruthy()
      expect(getByTestId('radioGroup-option-option3')).toBeTruthy()
    })
  })

  describe('Edge Cases', () => {
    test('handles empty options array', () => {
      const tree = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} options={[]} />
        </BasicAppContext>
      )

      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })

    test('handles single option', () => {
      const singleOption = [{ label: 'Only Option', value: 'only' }]

      const tree = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} options={singleOption} />
        </BasicAppContext>
      )

      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })

    test('handles selectedValue that does not match any option', () => {
      const { getByTestId } = render(
        <BasicAppContext>
          <RadioGroup {...defaultProps} selectedValue="nonexistent" testID="radio-group" />
        </BasicAppContext>
      )

      // All options should be unselected
      expect(getByTestId('radio-group-option-option1').props.accessibilityState.selected).toBe(false)
      expect(getByTestId('radio-group-option-option2').props.accessibilityState.selected).toBe(false)
      expect(getByTestId('radio-group-option-option3').props.accessibilityState.selected).toBe(false)
    })
  })
})
