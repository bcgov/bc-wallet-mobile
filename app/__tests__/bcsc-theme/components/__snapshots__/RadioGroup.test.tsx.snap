// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup Component Edge Cases handles empty options array 1`] = `
<View
  style={{}}
  testID="test-radio-group"
>
  <View
    style={
      {
        "paddingBottom": 4,
      }
    }
  />
</View>
`;

exports[`RadioGroup Component Edge Cases handles single option 1`] = `
<View
  style={{}}
  testID="test-radio-group"
>
  <View
    style={
      {
        "paddingBottom": 4,
      }
    }
  >
    <View
      accessibilityLabel="Only Option"
      accessibilityRole="radio"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": false,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
      testID="test-radio-group-option-OnlyOption"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "flex": 1,
            "fontFamily": "BCSans-Regular",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        Only Option
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#003366",
            "borderRadius": 12,
            "borderWidth": 3,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`RadioGroup Component Rendering renders correctly with default props 1`] = `
<View
  style={{}}
  testID="test-radio-group"
>
  <View
    style={
      {
        "paddingBottom": 4,
      }
    }
  >
    <View
      accessibilityLabel="Option 1"
      accessibilityRole="radio"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": false,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
      testID="test-radio-group-option-Option1"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "flex": 1,
            "fontFamily": "BCSans-Regular",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        Option 1
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#003366",
            "borderRadius": 12,
            "borderWidth": 3,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      />
    </View>
    <View
      style={
        {
          "height": 8,
        }
      }
    />
    <View
      accessibilityLabel="Option 2"
      accessibilityRole="radio"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": false,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
      testID="test-radio-group-option-Option2"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "flex": 1,
            "fontFamily": "BCSans-Regular",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        Option 2
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#003366",
            "borderRadius": 12,
            "borderWidth": 3,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      />
    </View>
    <View
      style={
        {
          "height": 8,
        }
      }
    />
    <View
      accessibilityLabel="Option 3"
      accessibilityRole="radio"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": false,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
      testID="test-radio-group-option-Option3"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "flex": 1,
            "fontFamily": "BCSans-Regular",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        Option 3
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#003366",
            "borderRadius": 12,
            "borderWidth": 3,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`RadioGroup Component Rendering renders correctly with mixed disabled options 1`] = `
<View
  style={{}}
  testID="test-radio-group"
>
  <View
    style={
      {
        "paddingBottom": 4,
      }
    }
  >
    <View
      accessibilityLabel="Option 1"
      accessibilityRole="radio"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": false,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
      testID="test-radio-group-option-Option1"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "flex": 1,
            "fontFamily": "BCSans-Regular",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        Option 1
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#003366",
            "borderRadius": 12,
            "borderWidth": 3,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      />
    </View>
    <View
      style={
        {
          "height": 8,
        }
      }
    />
    <View
      accessibilityLabel="Option 2 (Disabled)"
      accessibilityRole="radio"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": false,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
      testID="test-radio-group-option-Option2(Disabled)"
    >
      <Text
        style={
          {
            "color": "#606060",
            "flex": 1,
            "fontFamily": "BCSans-Regular",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        Option 2 (Disabled)
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#606060",
            "borderRadius": 12,
            "borderWidth": 3,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      />
    </View>
    <View
      style={
        {
          "height": 8,
        }
      }
    />
    <View
      accessibilityLabel="Option 3"
      accessibilityRole="radio"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": false,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
      testID="test-radio-group-option-Option3"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "flex": 1,
            "fontFamily": "BCSans-Regular",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        Option 3
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#003366",
            "borderRadius": 12,
            "borderWidth": 3,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`RadioGroup Component Rendering renders correctly with selected value 1`] = `
<View
  style={{}}
  testID="test-radio-group"
>
  <View
    style={
      {
        "paddingBottom": 4,
      }
    }
  >
    <View
      accessibilityLabel="Option 1"
      accessibilityRole="radio"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": false,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
      testID="test-radio-group-option-Option1"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "flex": 1,
            "fontFamily": "BCSans-Regular",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        Option 1
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#003366",
            "borderRadius": 12,
            "borderWidth": 3,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      />
    </View>
    <View
      style={
        {
          "height": 8,
        }
      }
    />
    <View
      accessibilityLabel="Option 2"
      accessibilityRole="radio"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": true,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
      testID="test-radio-group-option-Option2"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "flex": 1,
            "fontFamily": "BCSans-Regular",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        Option 2
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#003366",
            "borderRadius": 12,
            "borderWidth": 3,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#003366",
              "borderRadius": 6,
              "height": 12,
              "width": 12,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "height": 8,
        }
      }
    />
    <View
      accessibilityLabel="Option 3"
      accessibilityRole="radio"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": false,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
      testID="test-radio-group-option-Option3"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "flex": 1,
            "fontFamily": "BCSans-Regular",
            "fontSize": 18,
            "fontWeight": "normal",
          }
        }
      >
        Option 3
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#003366",
            "borderRadius": 12,
            "borderWidth": 3,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;
